fineract-provider/src/main/java/org/apache/fineract/portfolio/client/domain/ClientRepository.java

```diff
@@ -29,4 +29,35 @@ interface ClientRepository extends JpaRepository<Client, Long>, JpaSpecification
 
     @Query(FIND_CLIENT_BY_ACCOUNT_NUMBER)
     Client getClientByAccountNumber(@Param("accountNumber") String accountNumber);
+
+    @Query("""
+            SELECT client.id
+            FROM Client client
+            WHERE client.externalId = :externalId
+                """)
+    Long getClientIdByExternalId(String externalId);
+
+    @Query("""
+            SELECT client
+            FROM Client client
+            JOIN client.office office
+            LEFT JOIN client.transferToOffice transferToOffice
+            WHERE client.id = :clientId
+            AND (office.hierarchy LIKE :officeHierarchy OR transferToOffice.hierarchy LIKE :transferToOfficeHierarchy)
+                """)
+    Client fetchByClientIdAndHierarchy(Long clientId, String officeHierarchy, String transferToOfficeHierarchy);
+
+    @Query("""
```

Comment:
We dont need this. We should not fetch any data by the externalid, rather fetch the client id by the external id and leave untouched the rest of the logic to avoid duplications

Reply:
Done removed!